### PR TITLE
Make the InputContainer underline's visibility configurable

### LINKED ISSUE #6942: InputContainer underline should be optional
The newly introduced InputContainer is a great wrapper for standardizing labeling for input widgets. Its underline however should be optional as this is meant to be used with many types of input widgets such as dropdowns, sliders, checkboxes etc.

For instance using it with a slider currently causes this:

image

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes #6942

### CODE REVIEW

File: packages/flutter/lib/src/material/input.dart [hunk 1]
```diff
@@ -414,6 +429,7 @@ class Input extends StatefulWidget {
   /// Text to show inline in the input field when it would otherwise be empty.
   final String hintText;
 
+
```
Comment (abarth) [inline packages/flutter/lib/src/material/input.dart]: Seems spurious

**Comment by abarth** [comment]
Seems like a double negative, but maybe that's ok given that it defaults to false...

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/material/input.dart [hunk 2]
```diff
@@ -304,16 +308,26 @@ class _InputContainerState extends State<InputContainer> {
...
-      decoration: new BoxDecoration(
-        border: border,
-      ),
-      child: config.child,
-    ));
+    Widget divider;
+    if (config.hideDivider) {
+      divider = new Container(
+        margin: margin,
+        padding: padding,
```
Comment (abarth) [inline packages/flutter/lib/src/material/input.dart]: I think these two will be slightly different heights.  Is that what you intended?

Comment (HansMuller) [inline packages/flutter/lib/src/material/input.dart]: Wow, nice catch, that's not what I intended!

**Comment by HansMuller** [comment]
I agree that the default double negative, hideDivider: false, is a little off. I think we prefer boolean properties that default to false and there's the HideDropdownButtonUnderline as a precedent for a property named hideFoo.  I didn't add a new inherited widget because, so far, this flag seems to be something that will vary within a form.

**Comment by Hixie** [comment]
I'd rather we didn't go with the double negative thing. The Web had "disabled=true" and it caused no end of hassle.

I suggest a property `dividerStyle` with values from an enum: `DividerStyle.none`, `DividerStyle.solid`, defaulting to solid, not allowed to be null.

**Comment by Hixie** [comment]
(HideDropdownButtonUnderline is dubious at best, but at least it doesn't have a boolean value associated with it.)

**Comment by HansMuller** [comment]
DividerStyle seems reasonable however that may cause some confusion with the various ways of rendering the divider per focus, disabled, error: https://material.google.com/components/text-fields.html#text-fields-single-line-text-field.

**Comment by Hixie** [comment]
DividerVisibility: visible, hidden?

**Pull request merged**